include/nil/blueprint/parser.hpp

```diff
@@ -314,6 +314,14 @@ namespace nil {
                         assert(false && "zkml_batch_norm intrinsic is not implemented yet");
                         return false;
                     }
+                    case llvm::Intrinsic::expect: {
+                        var x = frame.scalars[inst->getOperand(0)];
+                        var y = frame.scalars[inst->getOperand(1)];
+                        bool res = var_value(assignmnt, x) == var_value(assignmnt, y);
```

Comment:
As I understand from the [documentation](https://llvm.org/docs/LangRef.html#llvm-expect-intrinsic) and [usage example](https://github.com/NilFoundation/zkllvm-circifier/blob/master/llvm/lib/CodeGen/IntrinsicLowering.cpp#L240), the result of this intrinsic must be the first operand.

Reply:
Sure, you're right, that's a mistake. Should we use `assignmnt.public_input` here and if yes, how to marshal `var` value into it?

Reply:
Done